Cluster Programs

```diff
@@ -1,3 +1,68 @@
+
+3) In the original dataset from Harvard Dataverse you can find the variables Invnum and Invnum_N.
+Invnum: Initial inventor number: Patent + Invseq.
+Invnum_N: Disambiguated inventor number. This function assign a number to each of the individual inventors, 
+this way an inventor with more than one patent will have the same number. 
+In the program, we are using Invnum_N, which later it’s rename to Invnum.
+
+4) Regarding the next part, this is a good page for figuring out which distribution or method we are looking for
+https://berkorbay.github.io/fe522/02_Monte_Carlo_Simulation.html
+Looking the documentation, it said that before choosing a distribution for running the iterations, I need to perform and analysis
+on my dataset first. In this page there is a good starting point. 
+https://www.epa.gov/sites/production/files/2014-11/documents/montecar.pdf
+
+I need to figure this out before use the Monte Carlo function in R.
+This means that, I need to have my merge_patents dataset ready to begin this step. I'm trying to make it work with the small data sets. 
+
+There is a method to calculate the value of Pi using a Monte Carlo, it uses the radius of a circumference and then randomly put dots in
+a square of length 1, then it measure how many dots are inside the circumference and how many outside. 
+I think this could work in our program, if we take the radius as the 5, 10 or 20 miles of the different clusters. 
+
+5) Regarding the R code. 
+I'm looking at this part, creating the t2 vector and replacing as the object in cites_mypatent. When I do this, it doesn't create a matrix, 
+of column x row as you mentioned in Frank's code. I have run the code in the small datasets and I don't get any repeating patent in the cited.
+I'm not sure if using the vector t2 in the cites_mypatent is the correct thing to do yet.
+But if put any additional restriction on the "out" condition the program send warnings of length. 
+
+big_citations <-as.data.table(smallpossiblenclass[,-c(1:9)])
+
+#this changes all NAs in big_citations2 to 0's 
+for(j in seq_along(big_citations)){
+  set(big_citations, i=which(is.na(big_citations[[j]])), j=j, value=0)
+}
+#Here we look run the condition that the cite1 to cite 11 are not in cited or in patent
+t2 <- which(big_citations[,!(smallmatching2$patent %in% big_citations) & (!(smallmatching2$cited %in% big_citations))])
+
+# Function that returns vector of all rows from big that match 'mypatent,' a list with named elements corresponding to small. 
+#(I.e. mypatent will be a row from small)
+
+get_matching_rows <- function(mypatent) {
+  
+  # Indicator (TRUE/FALSE) for patents from big that cite mypatent 
+  cites_mypatent <- unlist(lapply(t2, function(x) mypatent$patent %in% x))
+
+  out <- which((smallpossiblenclass$nclass == mypatent$nclass))
+  return(out)
+}
+For the past few days I've been looking at this part, trying with the rbinds and the data table documentation, 
+I'm sorry I have no better news. 
```

Comment:
Here, my idea was to write a bunch of statements like the ones that create t2--one for each of the conditions. Then one big statement, that should not, hopefully, require a function, that just finds which indices are in each of the vectors that were created by those condition-checking statements (say, t2, t3, t4, t5). I'm not sure what the command would be that would do that.

Maybe that won't work, and the better approach is to enlarge the step that creates t2 to include all of the conditions.

Reply:
I don't think we can include all the conditions in the step that creates t2, because t2 it's been created by big_citations, and big_citations only contains the cite information (cite1 to cite772), not even the patent number. 
So, if we implement the rest of the conditions, we would need to do it in the complete dataset (smallpossiblenclass) and that would change the skeleton of what Frank send us.

Reply:
I meant to have other steps like the one that creates t2 that outputs lists of indices that meet those other conditions. Then a final step that finds the indices that are in each of those lists. Each of the individual steps could use different data sets.

I am back in Syracuse if we need to meet. I could meet this afternoon before 3 if needed.

Reply:
If you can meet today, that would be good. 
What time would work for you?